packages/remirror__extension-link/src/link-extension.ts

```diff
@@ -603,29 +615,91 @@ export class LinkExtension extends MarkExtension<LinkOptions> {
             }
 
             const nodeText = tr.doc.textBetween(pos, pos + node.nodeSize, undefined, ' ');
-            this.findAutoLinks(nodeText)
+
+            const foundAutoLinks: FoundAutoLink[] = [];
+
+            for (const match of findMatches(nodeText, NON_WHITESPACE_REGEX)) {
```

Comment:
Instead of doing one pass to find auto links ( removed line 606 - "findAutoLinks" ) in the "nodeText" we split the "nodeText" into smaller pieces. 

From here we check for new and exciting links and possibly create, update or remove invalid links.